### DISCUSSION TITLE
Uncaught ReferenceError: Alpine is not defined

### DISCUSSION THREAD

**Discussion by ericxin1982** [alpinejs/alpine#3901] [1. Help]
![image](https://github.com/alpinejs/alpine/assets/9047436/91ad8700-509f-4ba3-8cf0-195163181a52)

Hi there,

I need some help, I try to practice reactive function on that way, but it seems not working, I am very new to alpine, I do not know how to start here.
I am using cdn this way.

Thanks
Eric Xin

**Comment by ekwoka** [comment]
Your code is running before Alpine runs.

You can make your script tag `type=module` to have it run after defered scripts before it.

also, don't forget to defer the `vimesh/ui`

**Reply by bokunodev** [reply]
note for future alpine user who stumbled upon the same problem, your js code should run before alpinejs. since all it did was registering an event handler. it does not make any sense to register event handler after the event happened.

**Reply by ekwoka** [reply]
You need code that accesses `Alpine` after Alpine exists.

An event listener is one way to do that.

The best is using modules an just importing Alpine.

**Reply by bokunodev** [reply]
using module, require me to use `bundler` right? importing alpine from `<script type="module">` wont do.

based on the [doc](https://alpinejs.dev/globals/alpine-data#registering-from-a-bundle) i need to call `Alpine.start`. but doing so throw a warning.

**Reply by ekwoka** [reply]
You can use script type module.

You just need to provide a valid URL, or have an import map of how to take a short name and map it to a URL.

**Reply by bokunodev** [reply]
i don't think import map has anything todo with this.

i'm confused. the module is loaded. but calling `Alpine.start` throws a warning "alpine already started. calling alpine.start multiple time  might cause a problem". so i assume, `Alpine.start` has already been called as side effect of the import.

based on the doc, section [Lifecycle](https://alpinejs.dev/essentials/lifecycle). alpine emits 2 events `alpine:init` which is emitted **before** `Alpine.start`. **but after the script has properly loaded** and `alpine:initialized` which is emitted **after** `Alpine.start`.

my alpine version is `3.14.1` (current latest stable release)

**Reply by ekwoka** [reply]
If it has already been called then it's because you have it loading via a script tag or are importing a self initialized version.

both of those events emit once `alpine.start` is called, `init` will just be before it actually processes anything on the DOM.
